xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -652,6 +675,12 @@ public URL createAttachmentRevisionURL(String filename, String spaces, String na
      */
     @Override
     public String getURL(URL url, XWikiContext context)
+    {
+        return getURL(url, context, null);
+    }
+
+    @Override
+    public String getURL(URL url, XWikiContext context, Map<String, String> queryParameters)
```

Comment:
The point of XWikiServletURLFactory#getURL is not to generate an URL but to move from absolute to relative URL (if applicable). It's really not it's job to add a query string the way it's currently defined.